Mesh_2/examples/Mesh_2/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.1...3.20)
 project(Mesh_2_Examples)
 
 find_package(CGAL REQUIRED)
-
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pthread" )
```

Comment:
This shouldn't go in the file. Flags are supposed to be given at configuration time, as a cmake option. For example, calling `-pthread` will most certainly make the whole Mesh_2 examples unusable on Windows.

Reply:
Thank you for this remark, I did not know this. Anyway, the example was added "just to be sure" I found a code that may segfault. The mesh_optimization raises the same error.

What should I do with this part of code ?

thank you again for your time

Yann

Reply:
- `-Wall` should not appear in the `CMakeLists.txt`. You can add it to your cmake user presets.
- As for `-pthread`, the CMake portable way to do it:
  ```cmake
  find_package(Threads)
  target_link_libraries(test_parallel_Mesh_global_optimizer PRIVATE Threads::Threads)
  ```
  and on Linux that will add `-pthread` and on other platform, that will modifies the compilation and linker flags with what is needed.

Reply:
Thank you for this comment.

Is there a way to add this link directive within the foreach ? The solution I found was to explicitely make the linking :
```cmake
file(
  GLOB cppfiles
  RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
  ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)
foreach(cppfile ${cppfiles})
  create_single_source_cgal_program("${cppfile}")
endforeach()

target_link_libraries(test_parallel_Mesh_global_optimizer PRIVATE Threads::Threads)
target_link_libraries(mesh_optimization PRIVATE Threads::Threads)
```

It may be enough for the debugging purpose.

Yann

Reply:
We do not want to add that in the `foreach`, to avoid linking all executables with `libpthread`. Only the two that require it are linked with threads.